src/ui/components/filters/sortoptionscomponent.js

```diff
@@ -104,8 +140,35 @@ export default class SortOptionsComponent extends Component {
     this.options[this.selectedOptionIndex].isSelected = false;
     this.options[optionIndex].isSelected = true;
     this.selectedOptionIndex = optionIndex;
-    this.showReset = this._config.showReset && optionIndex !== 0;
-    this.setState();
+  }
+
+  /**
+   * Set the 'checked' attribute for the selected option and remove it for all others
+   */
+  _updateCheckedAttributes () {
+    this.options.forEach((option, optionIndex) => {
+      const optionId = `#yxt-SortOptions-option_SortOptions_${optionIndex}`;
+      const optionEl = DOM.query(this._container, optionId);
+
+      if (this.selectedOptionIndex === optionIndex) {
+        optionEl && optionEl.setAttribute('checked', '');
+      } else {
+        optionEl && optionEl.removeAttribute('checked', '');
+      }
+    });
+  }
+
+  /**
+   * Set the visibility of the reset button based on this.showReset
```

Comment:
this is technically not setting the visibility, since visibility is a different css property than display